src/controllers/userProfileController.js

```diff
@@ -538,7 +538,9 @@ const userProfileController = function (UserProfile) {
       }
       if (
         req.body.infringements !== undefined &&
-        (await hasPermission(req.body.requestor, 'infringementAuthorizer'))
+        ((await hasPermission(req.body.requestor, 'addInfringements')) ||
+          (await hasPermission(req.body.requestor, 'deleteInfringements')) ||
+          (await hasPermission(req.body.requestor, 'editInfringements')))
```

Comment:
This will not limit each functionality to only their relevant modifications. (e.g. 'addInfringements' would also allow you to delete)

Reply:
For separating those functionalities, it might be easier to make them separate API endpoints.

Reply:
I've made changes. Please review again.

Reply:
Good job with the new APIs. Just need to remove this logic to not be duplicated. (or replace the `||`s with `&&`s)